policy: Sort the capture base on dependent relationship

cathay4t opened the pull request:
When child capture been placed before parent capture, nmstate will
complain about capture not found. Like this YAML:

```yml
child: interfaces.name==capture.child.routes.running.0.next-hop-interface
parent: routes.running.next-hop-interface=="eth1"
```

This patch will sort above YAML by placing capture after its parent capture
like:

```yml
parent: routes.running.next-hop-interface=="eth1"
child: interfaces.name==capture.child.routes.running.0.next-hop-interface
```

Implementation detail:
 * Each run of `set_capture_priority()`,
   `NetworkCaptureCommand::capture_priority` will be set to its parent's
   `capture_priority`. For standalone, set to 1. For parent not holding
   `capture_priority`, do nothing and wait next round by return
   false(indicate not finished).
 * Nmstate will run the `set_capture_priority()` up to 10 times allowing
   10 captures in any order.

Changed example policy `bridge-on-default-gw-no-dhcp` to unsorted
captures, the unit test case `test_policy_examples` will use it to test
this patch.

qinqon commented on rust/src/lib/policy/capture.rs:
```diff
@@ -86,6 +88,60 @@ impl NetworkCaptureRules {
+        Err(NmstateError::new(
+            ErrorKind::InvalidArgument,
+            format!(
+                "Failed to sort the policy capture in \
+                {SORT_CAPTURE_MAX_ROUND} round of rotation, \
+                please order the capture in desire \
+                policy by placing capture before its consumer"
```
User re-roder of the yaml dictionary makes no effect depending on the yaml parser implementation, the only thing that the user can check is the number of references or if there is a loop at them.

cathay4t replied:
New unit test included proving user sorted capture matters.

qinqon reviewed (changes requested):
Small nit on error message.

qinqon reviewed (approved):
Lgtm

cathay4t committed:
policy: Sort the capture base on dependent relationship

When child capture been placed before parent capture, nmstate will
complain about capture not found. Like this YAML:

```yml
child: interfaces.name==capture.child.routes.running.0.next-hop-interface
parent: routes.running.next-hop-interface=="eth1"
```

This patch will sort above YAML by placing capture after its parent capture
like:

```yml
parent: routes.running.next-hop-interface=="eth1"
child: interfaces.name==capture.child.routes.running.0.next-hop-interface
```

Implementation detail:
 * Each run of `set_capture_priority()`,
   `NetworkCaptureCommand::capture_priority` will be set to its parent's
   `capture_priority`. For standalone, set to 1. For parent not holding
   `capture_priority`, do nothing and wait next round by return
   false(indicate not finished).
 * Nmstate will run the `set_capture_priority()` up to 10 times allowing
   10 captures in any order.

Changed example policy `bridge-on-default-gw-no-dhcp` to unsorted
captures, the unit test case `test_policy_examples` will use it to test
this patch.

Signed-off-by: Gris Ge <fge@redhat.com>
